test/functional/legacy/prompt_buffer_spec.lua

```diff
@@ -4,150 +4,203 @@ local feed= helpers.feed
+      screen:expect([[
+        % -helx^                  |
+        ~                        |
+        ~                        |
+        ~                        |
+        [Prompt] [+]             |
+        other buffer             |
+        ~                        |
+        ~                        |
+        ~                        |
+                                 |
+      ]])
+      feed("<C-U>exit\n")
+      screen:expect([[
+        ^other buffer             |
+        ~                        |
+        ~                        |
+        ~                        |
+        ~                        |
+        ~                        |
+        ~                        |
+        ~                        |
+        ~                        |
+                                 |
+      ]])
+    end)
   end)
 
-  it('editing', function()
-    screen:expect([[
-      ^                         |
-      ~                        |
-      ~                        |
-      ~                        |
-      [Prompt]                 |
-      other buffer             |
-      ~                        |
-      ~                        |
-      ~                        |
-                               |
-    ]])
-    feed("i")
-    feed("hello<BS><BS>")
-    screen:expect([[
-      % hel^                    |
-      ~                        |
-      ~                        |
-      ~                        |
-      [Prompt] [+]             |
-      other buffer             |
-      ~                        |
-      ~                        |
-      ~                        |
-                               |
-    ]])
-    feed("<Left><Left><Left><BS>-")
-    screen:expect([[
-      % -^hel                   |
-      ~                        |
-      ~                        |
-      ~                        |
-      [Prompt] [+]             |
-      other buffer             |
-      ~                        |
-      ~                        |
-      ~                        |
-                               |
-    ]])
-    feed("<End>x")
-    screen:expect([[
-      % -helx^                  |
-      ~                        |
-      ~                        |
-      ~                        |
-      [Prompt] [+]             |
-      other buffer             |
-      ~                        |
-      ~                        |
-      ~                        |
-                               |
-    ]])
-    feed("<C-U>exit\n")
-    screen:expect([[
-      ^other buffer             |
-      ~                        |
-      ~                        |
-      ~                        |
-      ~                        |
-      ~                        |
-      ~                        |
-      ~                        |
-      ~                        |
-                               |
-    ]])
+  describe('(legacy test)', function()
+    before_each(function()
+      source([=[
+        func Test_prompt_buffer_getbufinfo()
```

Comment:
In the long run, it would be easier if you ported the entire `Test_prompt_buffer.vim` and modified it to run this test only. I don't mind if you literred most of the test cases there with `throw "skipped: TODO"`.